addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -2098,6 +2098,54 @@ QUnit.test("mentioned partners should not be notified if they are not member of
     assert.verifySteps(['message_post'], "the message should be posted");
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(2);
+
+    const uploadPromise = makeTestPromise(); // allows an attachment to be deleted before the completion of the upload
+    await this.start({
+        async mockFetch(resource, init) {
+            const _super = this._super.bind(this, ...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await uploadPromise;
+            }
+            return _super();
+        },
+    });
+    const composer = this.env.models['mail.composer'].create();
+    await this.createComposerComponent(composer);
+    const file1 = await createFile({
+        name: 'text1.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    const file2 = await createFile({
+        name: 'text2.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    await afterNextRender(() =>
+        inputFiles(
+            document.querySelector('.o_FileUploader_input'),
+            [file1, file2]
+        )
+    );
+    assert.containsN(
+        document.body,
+        '.o_Composer .o_Attachment',
+        2,
+        "should have two attachment"
+    );
+    await afterNextRender(() =>
+        document.querySelectorAll('.o_Attachment_asideItemUnlink')[1].click()
+    );
+    uploadPromise.resolve(); // simulates the completion of the upload of the first attachment
+    assert.containsOnce(
+        document.body,
+        '.o_Composer .o_Attachment',
+        "should have only one attachment"
```

Comment:
Clarify it's just the 1st uploading attachment, after having canceled uploading the 2nd attachment.